module/VuFind/src/VuFind/AjaxHandler/GetSideFacets.php

```diff
@@ -0,0 +1,300 @@
+use VuFind\Search\SearchRunner;
+use VuFind\Search\Solr\HierarchicalFacetHelper;
+use VuFind\Search\UrlQueryHelper;
+use VuFind\Session\Settings as SessionSettings;
+use Zend\Mvc\Controller\Plugin\Params;
+use Zend\View\Renderer\RendererInterface;
+
+/**
+ * "Get Side Facets" AJAX handler
+ *
+ * @category VuFind
+ * @package  AJAX
+ * @author   Ere Maijala <ere.maijala@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+class GetSideFacets extends \VuFind\AjaxHandler\AbstractBase
+    implements \Zend\Log\LoggerAwareInterface
+{
+    use \VuFind\Log\LoggerAwareTrait;
+
+    /**
+     * Recommend plugin manager
+     *
+     * @var RecommendPluginManager
+     */
+    protected $recommendPluginManager;
+
+    /**
+     * Search runner
+     *
+     * @var SearchRunner
+     */
+    protected $searchRunner;
+
+    /**
+     * Hierarchical facet helper
+     *
+     * @var HierarchicalFacetHelper
+     */
+    protected $facetHelper;
+
+    /**
+     * Main facet configuration
+     *
+     * @var \VuFind\Config\PluginManager
+     */
+    protected $facetConfig;
+
+    /**
+     * View renderer
+     *
+     * @var RendererInterface
+     */
+    protected $renderer;
+
+    /**
+     * Constructor
+     *
+     * @param SessionSettings         $ss       Session settings
+     * @param RecommendPluginManager  $rpm      Recommend plugin manager
+     * @param SearchRunner            $sr       Search runner
+     * @param HierarchicalFacetHelper $fh       Facet helper
+     * @param \Zend\Config\Config     $fc       Facet config
+     * @param RendererInterface       $renderer View renderer
+     */
+    public function __construct(SessionSettings $ss,
+        \VuFind\Recommend\PluginManager $rpm,
+        SearchRunner $sr, HierarchicalFacetHelper $fh,
+        \Zend\Config\Config $fc, RendererInterface $renderer
+    ) {
+        $this->sessionSettings = $ss;
+        $this->recommendPluginManager = $rpm;
+        $this->searchRunner = $sr;
+        $this->facetHelper = $fh;
+        $this->facetConfig = $fc;
+        $this->renderer = $renderer;
+    }
+
+    /**
+     * Handle a request.
+     *
+     * @param Params $params Parameter helper from controller
+     *
+     * @return array [response data, HTTP status code]
+     */
+    public function handleRequest(Params $params)
+    {
+        $this->disableSessionWrites();  // avoid session write timing bug
+
+        // Allow both GET and POST variables:
+        $request = $params->fromQuery() + $params->fromPost();
+
+        $results = $this->getFacetResults($request);
+        if ($results instanceof \VuFind\Search\EmptySet\Results) {
+            $this->logError('Faceting request failed');
+            return $this->formatResponse('', self::STATUS_HTTP_ERROR);
+        }
+
+        $recommend = $results->getRecommendations('side');
+        $recommend = reset($recommend);
```

Comment:
This appears to be always taking the first recommendation module as the rendering context... but what if SideFacetsDeferred is one of several configured options? What if there are multiple instances of SideFacetsDeferred, as previously discussed?

I tried setting up pubDateVisAjax above SideFacetsDeferred in my side recommendation area, and it didn't obviously break anything... so maybe I'm misunderstanding what is going on here... but this still looks potentially problematic, so I thought I'd better bring it up.

Reply:
True, it was a problem. Should be fixed now.

Reply:
@EreMaijala, I don't see a commit fixing this. Did you forget to push something up?

Reply:
I forgot to push the changes. Rebased and pushed, hopefully I didn't break anything.